app/routes/realtime_audio.py

```
@@ -0,0 +1,98 @@
+        self.last_chunk_time = None
+
+    def add_chunk(self, chunk):
+        self.buffer.append(chunk)
+        self.last_chunk_time = time.time()
+
+    def should_transcribe(self):
+        if len(self.buffer) >= self.max_buffer_size:
+            return True
+        if self.last_chunk_time and (time.time() - self.last_chunk_time) >= 0.5:
+            return True
+        return False
```

Ruff (0.11.9)

Return the condition directly

Inline condition

(SIM103)